script.js

```diff
@@ -21,94 +21,87 @@ document.addEventListener("DOMContentLoaded", () => {
+        el: "table",
+        styles: `
         width: 1200px;
         background: gray;
         padding: 20px;
         margin: 0 auto;
         text-align: center;
         border: 2px solid black;
         `
+      }
       );
 
-      const caption = createEl(
-        "caption",
-        tableCaption,
-        `
+      const caption = createEl({
+        el: "caption",
+        text: tableCaption,
+        styles:`
         color: gray;
         font-size: 30px;
         padding: 10px 0px;
         `
-      );
-      const thead = createEl("thead");
-      const trHead = createEl("tr");
-      const tbody = createEl("tbody");
+      });
+      const thead = createEl({el: "thead"});
+      const trHead = createEl({el: "tr"});
+      const tbody = createEl({el: "tbody"});
 
       table.append(caption);
       table.append(thead);
       table.append(tbody);
       thead.append(trHead);
 
-      thNames.forEach((element) => {
-        const th = createEl(
-          "th",
-          element,
-          `
+      thNames.forEach((title) => {
+        const th = createEl({
+          el:"th",
+          text: title,
+          styles:`
           font-size: 18px;
           text-transform: uppercase;
           padding: 10px 0px;
           `
-        );
+        });
         trHead.append(th);
       });
 
-      const data = await fetchedData;
+      const data = await fetchedData //!!!
 
       for (let item of data) {
-        const tr = createEl("tr");
-        const input = createEl("input");
-        input.type = "checkbox";
-        input.setAttribute("id", item.id);
-        const td = createEl("td");
-        td.append(input);
-        tr.append(td);
+        const tr = createEl({el:"tr"});
+        const td = createEl({el:"td", styles : `padding: 10px;`})
+        const input = createEl({el: "input"})
+        input.type = "checkbox"
+        input.setAttribute("id", item.id)
+        td.append(input)
+        tr.append(td)
         thNames.slice(1).forEach((thName) => {
-          const td = createEl("td", item[thName]);
+          const td = createEl({el:"td", text: item[thName], styles: `
+          padding: 10px;
+          `});
           tr.append(td);
         });
         tbody.append(tr);
       }
       document.body.prepend(table);
 
-      const allTd = document.querySelectorAll("td");
-      for (let i = 0; i < allTd.length; i++) {
-        allTd[i].style.cssText = `
-        padding: 10px;
-        `;
-      }
-    } catch (error) {
-      console.log(error.message);
     }
-  };
 
   const createBtn = () => {
-    const button = createEl(
-      "button",
-      "delete",
-      `width: 60px; height: 30px; margin: 20px auto; display: block; background: gray; border: none; color: white;`
-    );
+    const button = createEl({
+      el:"button",
+      text:"delete",
+      styles:`width: 60px; height: 30px; margin: 20px auto; display: block; background: gray; border: none; color: white;`
```

Comment:
Add a coma for the last property in the object or array.
Configure the VSCode for it